pkg/server/sotw/v2/server.go

```diff
@@ -302,89 +215,16 @@ func (s *server) process(stream Stream, reqCh <-chan *discovery.DiscoveryRequest
 				}
 			}
 
-			// cancel existing watches to (re-)request a newer version
-			switch {
-			case req.TypeUrl == resource.EndpointType:
-				if values.endpointNonce == "" || values.endpointNonce == nonce {
-					if values.endpointCancel != nil {
-						values.endpointCancel()
-					}
-					values.endpoints, values.endpointCancel = s.cache.CreateWatch(req)
-				}
-			case req.TypeUrl == resource.ClusterType:
-				if values.clusterNonce == "" || values.clusterNonce == nonce {
-					if values.clusterCancel != nil {
-						values.clusterCancel()
-					}
-					values.clusters, values.clusterCancel = s.cache.CreateWatch(req)
-				}
-			case req.TypeUrl == resource.RouteType:
-				if values.routeNonce == "" || values.routeNonce == nonce {
-					if values.routeCancel != nil {
-						values.routeCancel()
-					}
-					values.routes, values.routeCancel = s.cache.CreateWatch(req)
-				}
-			case req.TypeUrl == resource.ListenerType:
-				if values.listenerNonce == "" || values.listenerNonce == nonce {
-					if values.listenerCancel != nil {
-						values.listenerCancel()
-					}
-					values.listeners, values.listenerCancel = s.cache.CreateWatch(req)
-				}
-			case req.TypeUrl == resource.SecretType:
-				if values.secretNonce == "" || values.secretNonce == nonce {
-					if values.secretCancel != nil {
-						values.secretCancel()
-					}
-					values.secrets, values.secretCancel = s.cache.CreateWatch(req)
-				}
-			case req.TypeUrl == resource.RuntimeType:
-				if values.runtimeNonce == "" || values.runtimeNonce == nonce {
-					if values.runtimeCancel != nil {
-						values.runtimeCancel()
-					}
-					values.runtimes, values.runtimeCancel = s.cache.CreateWatch(req)
-				}
-			default:
-				typeUrl := req.TypeUrl
-				responseNonce, seen := values.nonces[typeUrl]
-				if !seen || responseNonce == nonce {
-					// We must signal goroutine termination to prevent a race between the cancel closing the watch
-					// and the producer closing the watch.
-					if terminate, exists := values.terminations[typeUrl]; exists {
-						close(terminate)
-					}
-					if cancel, seen := values.cancellations[typeUrl]; seen && cancel != nil {
-						cancel()
+			typeUrl := req.TypeUrl
+			responseNonce, seen := values.nonces[typeUrl]
+			if !seen || responseNonce == nonce {
+				if cancel, seen := values.cancellations[typeUrl]; seen && cancel != nil {
+					cancel()
+					if err := processAll(); err != nil {
```

Comment:
Why is processAll guarded by an individual type URL? What about other muxed type URLs?

Reply:
`processAll` will process all types that appear in responses channel, it just happens that if we are talking about synchronous `cancel` - to guarantee that there is a place in buffer - we can only drain if we cancel